### PR TITLE
⚒️ Integrate foundry/forge

### PR DESCRIPTION
**Pull request by tommyrharper**
PR to integrate foundry/forge into the repo.

- [x] Setup Forge
- [x] Add initial unit test
- [x] Setup VSCode extension file linking
- [x] Updated hardhat config
- [x] Fixed hardhat CI tests
- [x] Setup Forge tests running in CI

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -26,6 +27,18 @@ task("accounts", "Prints the list of accounts", async (args, hre) => {
     }
 });
 
+// Add a subtask that sets the action for the TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS task
+// Add a subtask that sets the action for the TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS task
```
Comment (JaredBorders) [inline hardhat.config.ts:31]: Must've pasted this twice

**Review by JaredBorders — APPROVED** [review]
Nice work! Glad you were able to knock this out quickly. I left a single comment where you had duplicated a comment. Otherwise LGTM

**Comment by JaredBorders** [comment]
One small thing to consider: maybe add gas snapshots. I did this manually for the margin-manager repo, but keeping track of gas changes is helpful. I think it only works with forge tests, so might not be useful now, but will be in the future 👍
-
`"gas-snapshot": "forge snapshot"` for reference

File: hardhat.config.ts (hunk 1 above)
Comment (tommyrharper) [reply hardhat.config.ts:31]: nice catch! sorted.

**Pull request merged**